pkg/util/ec2/ec2_tags.go

```diff
@@ -127,6 +128,13 @@ func getTagsWithCreds(ctx context.Context, instanceIdentity *ec2Identity, awsCre
 	}
 
 	connection := ec2.New(awsSess)
+
+	// We want to use 'ec2_metadata_timeout' here instead of current context. 'ctx' comes from the agent main and will
+	// only be canceled if the agent is stopped. The default timeout for the AWS SQK is 1 minutes (20s timeout with
+	// 3 retries). Since we call getTagsWithCreds twice in a row, it can be a 2 minutes latency.
+	ctx, cancel := context.WithTimeout(ctx, config.Datadog.GetDuration("ec2_metadata_timeout")*time.Millisecond)
```

Comment:
Is 300ms enough to pull this data? Seems like it would be a huge reduction in wait time and I wonder if slow machines would be able to get the tags under those constraints.

Reply:
I'm currently testing this before merging.!
In all cases we would increase the ec2 timeout rather than come back to the default one from the SDK.

Reply:
Hmm, how so if we're overriding the SDK timeout of 20s w/ 0.3s?

Reply:
I mean we would increase `ec2_metadata_timeout` value if needed.

Reply:
Sure but as-is we're not making that increase in this PR. Overall, I just want us to be fairly sure not to have a regression due to this timeout change on slow machines.

Reply:
I finished testing this. We can collect tags on both windows and linux with t2.micro instances without any issue.